mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -103,11 +105,13 @@ public void preCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable)
       }
     }
 
+    Set<String> identifierFieldNames = HiveIcebergSerDe.getIdentifierFieldSet(
+        hmsTable.getParameters().remove(InputFormatConfig.IDENTIFIER_FIELD_NAMES));
```

Comment:
Could we use just a `get` here, and remove the parameter using the `PARAMETERS_TO_REMOVE`?

Reply:
Also we might want to move this into the `schema` method